pkg/ingress/controller.go

```diff
@@ -504,6 +516,85 @@ func (c *Controller) syncConsumer(ctx context.Context, consumer *apisixv1.Consum
 	}
 	return
 }
+
+func (c *Controller) syncEndpoint(ctx context.Context, ep kube.Endpoint) error {
```

Comment:
Why write sync in the controller?

Reply:
Because these logics can be reused by both the endpoint controller and endpointslice controller.

Reply:
I mean put them in another file maybe better , not in controller .